### ISSUE TITLE
Nara Meeting Minutes

### ISSUE THREAD

**Issue by bedeho** [Joystream/joystream#4736]
**MM 26.04.23**

# Status

Migration to Substrate version is pretty muchd one, only weight generation is left. No changes were needed to any palletss in terms of type signatures, storage representations or business logic, it was only renaming a bunch of types for the most.
- All integration tests are passing without need for any change, which means CLI, Hydra, QN, Colossus, Argus and Validator node are all building, running and basically working.
- Polkadotjs version did not need to be changed, hence everything indicates Atlas, Pioneer will also keep working.
- New `nara` branch with this work has been created.
- We still need our Substrate fork, because there is a lingering change to staking pallet which upstream substrate does not have, but @mnaamani will open a PR to upstream to see if they will accept it.

## Migrations

- No on-chain mirgation of Joystream pallets is needed.
- **Multisig pallet dose however need multiple migrations, all executing in one block**

# TODO
- @ignazio-bovo should try to update his creator token work by pulling in new this new work, incorporating, and merging into `nara`. CRTs should still be filtered in `Nara`, but latest implementation should be there. This work can start now. https://github.com/Joystream/joystream/issues/4747
- @mnaamani will open a PR to upstream Substrate to see if they will accept it our changes to staking pallet.
- @mnaamani will complete benchmarking
- @mnaamani will start to review #4589, with an eye towards identifying whether migrations will be needed, as business logic change is trivial.
- @bedeho will complete release plan draft
- We will start having weekly calls again, although with skeleton crew, implications for other teams is minimal  now.

**Comment by bedeho** [comment]
# Meeting

Wednesday 13 September 2023.

# Status

- We had a meeting last Wednesday, and there we concluded to
  - not include any CRT runtime changes in this release.
  - add a few minor tweaks on the runtime, mainly having to do with parameter tuning
- We still have not made progress on : staking pallet which upstream substrate does not have, but @mnaamani will open a PR to upstream to see if they will accept it.
- We rederived a new ETA, which was full deployment done on November 23rd. A lot of this time is being caused by JIP process being introduced properly, not technical work, which is mostly done.
- We tried to do the call on Discord, but this failed for logistical reasons - however the call was open, and this was a success, with meaningful exchange of information between release team and DAO.
- Next Monday, and onwards, we will do open calls on Discord.
- How exactly QA will be done, whether led by Martin or Builders, is not yet determined.
- We had no discussion about JIP process itself, this is being lead by Bedeho.
- We added new issues to runtime scope, but very easy ones
  - https://github.com/Joystream/joystream/issues/4866
  - https://github.com/Joystream/joystream/issues/4852
  - https://github.com/Joystream/joystream/issues/4822

# Milestone ETAs

- First staging network: Sep 21.
- Feature Frozen: Sep 29.
- JIP1 finalied: Sep 29.
- JIP2 finalized: October 13.
- JIP3 finalized: October 31.
- **Upgrade executed**: Nov 23.


## Risks

1. It is still not understood what migrations are needed for: https://github.com/Joystream/joystream/issues/4589
2. It is still not understood what is cusing Hydra issues on runtiem upgrade: https://github.com/Joystream/joystream/issues/4741

**Comment by bedeho** [comment]
# Meeting

Monday 18 September 2023.

# Attendance

- `Thesan`
- `mrbovo.eth`
- `bedeho`
- `freakstatic`
- `zeeshanakram`

# Status

- Monday was done on Discord successfully.
- Still unclear 
- Work on QN, CLI is done, revie pending.
- `freakstatic` has already done first runtime task w.r.t. forum depth, but did it on wrong branch (master), is going to open on `Nara` again soon.
- `freakstatic` also was ahppy to take on all other runtime tasks on this list from `mokhtar`, and will do them in separate PRs
  - https://github.com/Joystream/joystream/issues/4866 
  - https://github.com/Joystream/joystream/issues/4852
  - https://github.com/Joystream/joystream/issues/4870
  - https://github.com/Joystream/joystream/issues/4822
- Despite heavy other prios, `mrbovo.eth` is going to try to unblock reviews that are going to end up blocking first staging network.
- JIP progress is limited.
- QA work leadership still not specified from DAO side.
- DAO side has expressed an interest in taking on more devops work that `mokhtar` has been doing.

# Milestone ETAs

- First staging network: Sep 29.
- Feature Frozen: Sep 29.
- JIP1 finalied: October 3.
- JIP2 finalized: October 13.
- JIP3 finalized: October 31.
- **Upgrade executed**: Nov 23.


## Risks

1. It is still not understood what migrations are needed for: https://github.com/Joystream/joystream/issues/4589
2. It is still not understood what is causing Hydra issues on runtiem upgrade: https://github.com/Joystream/joystream/issues/4741
3. Still no QA leadership.
4. JIP process is lagging.

**Comment by bedeho** [comment]
# Meeting

Wednesday 20 September 2023.

# Attendance

- `Thesan`
- `mokhtar`
- `chaos77`
- `bedeho`
- `freakstatic`
- `zeeshanakram`
- `codefikeyz`
- `0x2bc`
- `yasir`

# Status

- We did not get it working on Discord, so it was on Zoom.
- Chaos confirms DAO will take on QA and QA leadership, with review support from Martin! Bedeho will provide an example of past QA plans. **Next meeting we need to 
- DAO will not take on devops side just yet, this will be done in separate exercise
- Freakstatic will take on integration testing work in his runtime PRs, to make sure there is some testing of new values, where test cases will include new hard coded values.
- A new issue was identified with how Hydra deals with upgrade where events are deprecated, and a fix is required unless we are prepare to keep around the unused events for curator deletion in our runtime. Zeeshan will fix: https://github.com/Joystream/joystream/issues/4877 
- Mokhtar will also create another issue for another beneficial Hydra fix.
- JIP progress is limited.

# Milestone ETAs

No milestones affected.

- First staging network: Sep 29.
- Feature Frozen: Sep 29.
- JIP1 finalied: October 3.
- JIP2 finalized: October 13.
- JIP3 finalized: October 31.
- **Upgrade executed**: Nov 23.

## Risks

1. It is still not understood what is causing Hydra issues on runtime upgrade: https://github.com/Joystream/joystream/issues/4741
2. JIP process is lagging.

**Comment by bedeho** [comment]
# Meeting

Monday 25 September 2023.

# Attendance

- `thesan`
- `mokhtar`
- `bedeho`
- `freakstatic`
- `zeeshanakram`
- `ivan`
- `jen`
- `mrbovo`
- `tomato`
- `leet_joy`

# Status

- First proper team wide call on Discord!
- This Hydra enhancement was reviewed and skipped from Nara scope: https://github.com/Joystream/hydra/issues/530 
- Chaos was unfortunatley not on the call, so we could not check up on QA planning progress, but here is a past QA plans that can be useful references: https://github.com/Joystream/joystream/issues/4576#issue-1557772798
- We decided to include a distinct working group max worker count constant per group: https://github.com/Joystream/joystream/issues/4822#issuecomment-1733369147 
- This runtime change was removed from Nara scope due to ambiguiity in process and merits. Is to be discusesd later: https://github.com/Joystream/joystream/issues/4870
- Freakstatic still working on : https://github.com/Joystream/joystream/issues/4866
- Mohtar to prepare issue with all migrations needed due to substrate update.
- JIP progress is limited.

# Milestone ETAs

No milestones affected.

- First staging network: October 1.
- Community QA: October 10th.
- Feature Frozen: October 19th.
- JIP1 finalied: October 29.
- JIP2 finalized: October 13.
- JIP3 finalized: October 31.
- **Upgrade executed**: Nov 23.

## Risks

1. It is still not understood what is causing Hydra issues on runtime upgrade: https://github.com/Joystream/joystream/issues/4741
2. JIP process is lagging.

**Comment by bedeho** [comment]
# Meeting

Thursday 28 September 2023.

# Attendance

- `thesan`
- `mokhtar`
- `bedeho`
- `freakstatic`
- `zeeshanakram`
- `ivan`
- `mrbovo`
- `tomato`
- `chaos77`
- `codefikeyz`
- `vikan`
- `yasir`
- `codefikeyz`
- `leet_joy`

# Status

- We will include CRT v1 release in this release, hence the CRT issuing extrinsic in the content pallet mus be unfiltered: https://github.com/Joystream/joystream/issues/4895
- Since the CRT pallet will be a lot of new functionlity, we will also take the opportunity to start the work of enabling 
  - https://github.com/Joystream/joystream/issues/4895
  - https://github.com/Joystream/joystream/issues/4897
  - https://github.com/Joystream/joystream/issues/4899
  - https://github.com/Joystream/joystream/issues/4898
  - https://github.com/Joystream/joystream/issues/4901
  - https://github.com/Joystream/pioneer/issues/4558
- Bedeho & Mokhtar will both fully audit CRT pallet: https://github.com/Joystream/joystream/issues/4900
- There will hopefully be a JIP video explainer recorded next week, and then a Discord meeting will be schedule to discuss and explain further for the benefit of council.
- Mokhtar will make issue about CLI changes needed due to multisig changes.
- Mokhtar will write requirements for a new alternative fast (1s block time) runtime configuration: https://github.com/Joystream/joystream/issues/4894
- QA will use two separate networks, a fast (1s block time) network and a normal speed networks, where the former is about testing with real (in terms of block length) period values.

# Milestone ETAs

- First staging network: October 12.
- Community QA: October 16.
- Feature Frozen: October 24.

JIP:
- JIP1 finalied: October 6.
- JIP2 finalized: October 27.
- JIP3 finalized: November 13.

- **Upgrade executed**: Nov 23.

## Risks

JIP process is lagging.

**Comment by bedeho** [comment]
# Meeting

Monday 2 October 2023.

# Attendance

- `thesan`
- `mokhtar`
- `bedeho`
- `freakstatic`
- `zeeshanakram`
- `mrbovo`
- `tomato`
- `codefikeyz`

# Status

- We decided to go for a pallet free approach to freezing existing pallets, rather than adding a new dedicated pallet for holding what pallets are frozen. This dramatically simplied the implementation hurdle for a new runtime dev like `freakstatic`, and the downsides were very limited. Change reflected here: https://github.com/Joystream/joystream/issues/4897#issuecomment-1742799791 
- `mrbovo` is unavailable 5-16 and will make sure to complete all remaining tasks that may block launching first staging network, in particular: https://github.com/Joystream/joystream/issues/4904
- `mrbovo` will also join the full CRT audit, on top of `bedeho` and `mokhtar`.
- We need to establish whether DAO can take on the following in a timely manner, but `chaos77` was not present
  - https://github.com/Joystream/joystream/issues/4903
  - https://github.com/Joystream/pioneer/issues/4557 
- It may be that we should upgrade Substrate yet one more time for Nara network, for a newer version, `mokhtar` will prepare issue for next call on Wed.
- `mokhtar` identified that we need to bump transaction version : https://github.com/Joystream/joystream/issues/4902 
- A new risk identified is that if we find a new issue in CRT pallet after audit, then this will delay community QA or require repition, which will have a cascading effect.
- Community QA must also cover CLI changes, such as new multisig changes (https://github.com/Joystream/joystream/issues/4878) or moderation changes.

# Milestone ETAs

Tech:
- First staging network: October 23.
- Community QA: October 30.
- Feature Frozen: Nov 6.

JIP:
- JIP1 finalied: October 6.
- JIP2 finalized: October 27.
- JIP3 finalized: November 13.

- **Upgrade executed**: Nov 23.

## Risks

JIP process is lagging.

**Comment by bedeho** [comment]
# Meeting

Wednesday 4 October 2023.

# Attendance

- `mokhtar`
- `chaos77`
- `bedeho`
- `freakstatic`
- `mrbovo`
- `tomato`

# Status

- Chaos was present and able to confirm that builders will take on
  - https://github.com/Joystream/joystream/issues/4903
  - https://github.com/Joystream/pioneer/issues/4557
- Chaos will revise the testing plan to be more focused around step by step descriptions of actions and observations to be verified, and this will require more detailed knowledge of changes which have been made.
- It may be that we should upgrade Substrate yet one more time for Nara network, for a newer version, `mokhtar` will prepare issue for next call on Wed.
- We clarified that scope of freezing for Nara is only `project_token` related functionality, nothing else, evne though freezing is going to be a long term feature. Added remark here: https://github.com/Joystream/joystream/issues/4897#issuecomment-1746630362
- We decided that callers to frozen pallets should be defensive, so they must check that hte pallet is frozen or not
- We bumped this to next week: It may be that we should upgrade Substrate yet one more time for Nara network, for a newer version, mokhtar will prepare issue for next call on Wed.
- Mokhtar identified that a type change in dispatch result type, in particular weight, is changed, so old full nodes will fail if extrinsics are submitted through them as an RPC noded (https://github.com/Joystream/joystream/issues/4907). Testing this must be part of QA, and relase of the full node must be in due time before runtime upgrade proposal is submitted.

# Milestone ETAs

Tech:
- First staging network: October 23.
- Community QA: October 30.
- Feature Frozen: Nov 6.

JIP:
- JIP1 finalied: October 6.
- JIP2 finalized: October 27.
- JIP3 finalized: November 13.

- **Upgrade executed**: Nov 23.

## Risks

JIP process is lagging.

**Comment by bedeho** [comment]
# Meeting

Monday 16th October 2023.

# Attendance

- `mokhtar`
- `chaos77`
- `bedeho`
- `freakstatic`
- `zeeshan`

# Status

- QA team still on standby waiting for support and clarificaiton of timeline for when Atlas CRT is ready.
- Mokhtar found `trie-db` 0.27.1 version bump in a recent Substrate PR did not fix pallet nemaing migration issue described prior. It also turns out that this `trie-db` version is actually not used in Substrate 0.9.39-1, it used in 0.9.42.
- Mokhtar selected 0.9.41, which also does not use trie-db fix: https://github.com/Joystream/joystream/pull/4932,  there were two dependency changes, `trie-db` 0.27.1 and wasmtime 6.0.2, Mokhtar will review again if the latter is truly needed to change from 6.0.1 which is used by dfeault in 0.9.41.
- The pallet migration issue turned out to be that post-upgrade hook assumed the migration code, as immplemented by the move pallet, cleaned up prefix even though it did not. This a captured in an assertion, but if that assertion is removed, then renaming pallets still works properly. Curiously, even if we use clear prefix, the assertion does indeed still fail. Since this last observation conflicts with our primary hypothesis for what is gong on, we will do two things
  - Mokhtar will approach Substrate core team to get their input, so we can make sure we have the correct understand. On stackoverflow for substrate and a whatever other channels exist.
  - QA will be done with a testnetwork with multiple, minimum 3, as in distinct host, validators, this should be convention in general (JIP2 wink wink). There is some ansible automation availbale to help with this, but it will require more assistance probably.
- There is some minimial documentation about how to stand up such a testnet, but Mokhtar will enhance.
- A note to QA: the QA must include the upgrade itself, not just starting on Nara.
- https://github.com/Joystream/joystream/issues/4928: When Martin originally made cli multisig, he used a package txwrapper, which depends on polkadot-js, this is now causing a version conflict. Builders will need to resolve this, possibly by no longer using this package.
- It is unclear whehter we need to itnroduce existential deposit in `project_token` pallet for revenue split treasury account, Ignazio was under the impression it would be needed due to in not getting setup in genesis block, but we are not sure. Answer can be verified by chekcing on-chain balance as last resort.

# Milestone ETAs

Tech:
- First staging network: October 27.
- Community QA: Nov 4.
- Feature Frozen: Nov 10.

JIP:
- JIP1 finalied: October 23.
- JIP2 finalized: October 28.
- JIP3 finalized: November 13.

- **Upgrade executed**: Nov 23.

## Risks

JIP process is lagging.

**Comment by bedeho** [comment]
# Meeting

Wednesday 18th October 2023.

# Attendance

- `mokhtar`
- `ignazio`
- `chaos77`
- `bedeho`
- `freakstatic`
- `zeeshan`

# Status

- Atlas ready with CRT October 30.
- Mokhtar we still on `0.9.41`, and he recommends we use wasmtime 6.0.2 because it included a security fix, so we will use in our fork. We will also stick to `trie-db` 0.27.1 because it fixes a problm we are likely to otherwise confront in the future.
- Mokhtar has created an issue on substrate StackOverflow, https://substrate.stackexchange.com/questions/10232/move-pallet-what-is-correct-way-to-verify-old-pallet-storage-is-deleted, he will also ask on Github as second backup.
- Ignazio has concluded that there is already an existential deposit set for project_token revenue split treasury pallet: https://github.com/Joystream/joystream/issues/4936
- We will not add support for specifying split amount for revenue splits in creator tokens, despite Atlas having this in the designs: https://github.com/Joystream/joystream/issues/4934 
- Mokhtar created better docs and interface for chainspec builder https://github.com/Joystream/joystream/pull/4935, will also backport to master branch, not only nara.
- New QA plan completed: https://github.com/Joystream/joystream/issues/4878#issuecomment-1764068306
- QA require two testnets, builders will be prepared to build and deploy networks.

# Milestone ETAs

## Tech

- First staging network: October 27.
- Community QA: Nov 4.
- Feature Frozen: Nov 10.

## JIP

- JIP1 finalied: October 23.
- JIP2 finalized: October 28.
- JIP3 finalized: November 13.

## Upgrade Executed

`Nov 23`

## Risks

JIP process is lagging.

**Comment by bedeho** [comment]
Attendance
@mokhtar 
@mrbovo.eth 
@freakstatic 

QA Plan
It’s necessary to ask @bwhm to review the new updates
@freakstatic got funding for the servers from @Chaos77  and will probably deploy a testnet this week.

Freeze pallet PR
Review was completed (we are just waiting for the pipeline to running the tests on the last commit) and will be probably merged soon
New PR with the proposal part will be created today (23/10) 

Multisig PR
Was reviewed but needs to be corrected (check @mokhtar  comments) - @Chaos77 

Creator Token Mappings
Almost done, some small fixes still needed - @mrbovo.eth 

Pioneer Nara branch
It seems that right now there isn’t any Nara branch, should it be created? 

Nara runtime types
This week @mokhtar plans to get more PRs merged and generate a new version for the nara runtime types. This will be necessary to add support for the freeze pallet support on pioneer.

**Comment by bedeho** [comment]
# Meeting

Wednesday 25th October 2023.

# Attendance

Active participants were

- `mokhtar`
- `ignazio`
- `chaos77`
- `bedeho`
- `freakstatic`
- `tomato`

# Status

- Chaos77 will consult with updated timeline on Atlas CRT availability.
- We have no reply on **StackExchange** for Substrate
https://substrate.stackexchange.com/questions/10232/move-pallet-what-is-correct-way-to-verify-old-pallet-storage-is-deleted We should try to get their attention in some other way later.
- Polkadotjs upgrade should be ugprade to use target v9.14.2 2 https://github.com/polkadot-js/api/releases/tag/v9.14.2 in parlalele work to upgrade across all apps and infra. But we will take out of Nara scope, because doing so across touch applications and infra, so it's risky to do now. An implication is that multisig updating of CLI is not part of Nara scope, and thus also QA.
- QN work can begin now already.

# Milestone ETAs

## Tech

- First staging network: Nov 2.
- Community QA: Nov 8.
- Feature Frozen: Nov 14.

## JIP

- JIP1 finalied: October 23.
- JIP2 finalized: October 28.
- JIP3 finalized: November 13.

## Upgrade Executed

`Dec 14`

## Risks

JIP process is lagging.

**Comment by bedeho** [comment]
# Meeting

Wednesday 1st November 2023.

# Attendance

Active participants were

- `mokhtar`
- `ignazio`
- `chaos77`
- `bedeho`
- `freakstatic`
- `tomato`

# Status

- Chaos left remarks
  - dmitry said crt feature enabled atlas instance for QA testing will be ready early next week
  - freakstatic will deploy testnets before the end of week
- Tomato will follow up on this issue in the Polkadot watercooler on Matrix: 
https://substrate.stackexchange.com/questions/10232/move-pallet-what-is-correct-way-to-verify-old-pallet-storage-is-deleted We should try to get their attention in some other way later.
- Mokhtar takes over responsibility to finish QN, so zeeshan can focus on other things: https://github.com/Joystream/joystream/issues/4898
- Some improvemnets were suggested about QA plan to help align us all on how QA networks will be setup: https://github.com/Joystream/joystream/issues/4878#issuecomment-1788774218 
- When does Pioneer work begin?

# Milestone ETAs

## Tech

- First staging network: Nov 7.
- Community QA: Nov 13.
- Feature Frozen: Nov 20.

## JIP

- JIP1 finalied: October 23.
- JIP2 finalized: October 28.
- JIP3 finalized: November 13.

## Upgrade Executed

`Dec 28`

## Risks

JIP process is lagging.

**Comment by mnaamani** [comment]
# Meeting

Monday November 6th 2023

# Attendance

- mokhtar
- freakstatic 
- Chaos77 
- $Mayor Tee.$JOY 
- Codefikeyz 
- spat_sochi 
- leet_joy

# Status

### Freeze pallet proposal status
- Merged, @freakstatic finished the runtime freeze proposal support and @mokhtar  added the support of QN.
@TheGrimSavage  will start to work on pioneer support of the freeze proposal. 

- @mokhtar  reported that it will be necessary to update the version of the polkadot-js on the pioneer dependencies. check https://github.com/Joystream/pioneer/pull/4552 is not the latest but it would work

### Deploy of the Nara testnets
The server setup was started but is not yet completed, scheduled to be completed today (06/11/2023)
The ephesus testnet with a block production of 1s will be deployed after the testnet with the Nara runtime.

**Comment by bedeho** [comment]
# Meeting

Monday 13st November 2023.

# Attendance

Active participants were

- `thegrimsavage`
- `mokhtar`
- `freakstatic`
- `tomato`
- `leet_joy`
- `0x2bc`
- `jen`

# Status

- Testnet has been launched on `Ephesus` but QA has not started.
- Pioneer development is just about done: https://github.com/Joystream/pioneer/pull/4625
- Unfortunately we had no input from QA lead, as `chaos77` was not able to participate due to connection issues.
- Current and future likely council reps largely were aware of JIP1, but not intimately, and would appreciate an explainer video and a call.

# Milestone ETAs

## Tech

- Community QA: Nov 8.
- Feature Frozen: Nov 20.

## JIP

- JIP1 finalied: November 24
- JIP2 finalized: December 6
- JIP3 finalized: December 12

## Upgrade Executed

`Dec 28`

## Risks

JIP process is lagging.

**Comment by bedeho** [comment]
Meeting was cut short, mokhtar and chaos were not there, not much could be covered, but seems like QA has started and is making progress.
